Automattic/WP-Job-Manager
# Admin notice modernisation

Fixes #2382

### Changes proposed in this Pull Request

* Different styling for different notice types added (error, warning, info, success)
* Setup for different notice classes in place
* Setup for icon retrieval in place
* Setup for Sensei's notice dismissal JS in place
* Setup for `get_dissmised_notices_`, `saved_dismissed_notices`, and `handle_notice_dimiss` is in place.

After discussing with @aaronfc about the scope of #2382 , we determined that setting up this PR, and the issue in question, would entail (1) firming up admin styling while (2) setting up persistent notice dismissal in a following issue. This was determined after realising that it would be a mark larger task than initially realised to have notice dismissal be persistent while also moving over all notice types to the new implementation.

To that end, this PR, like the issue it fixes:

* Adds support for different WPJM notice styling in admin
* Provides a foundation for notices to be migrated to the new notice design
* Makes notices dismissable, with persistence to be added in a latter, separate issue

### Testing instructions 

* Check out branch
* Ensure that you have an older version of a paid WPJM add-on active. If not, use a discount code and grab a license for an addon from https://wpjobmanger.com, then download an older copy of the addon from GH.
* You should then see a notice like so:

![notice](https://user-images.githubusercontent.com/2500940/232040947-aba405cb-b75c-4d2b-9f71-b81ffe468004.png)
